src/core/request_handler.rs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) 2020, Arm Limited, All Rights Reserved
+// SPDX-License-Identifier: Apache-2.0
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//          http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+// WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#![allow(dead_code)]
+
+use parsec_interface::requests::{Request, Response, Result};
+use std::os::unix::net::UnixStream;
+use std::path::PathBuf;
+use std::time::Duration;
+
+const DEFAULT_MAX_BODY_SIZE: usize = 1 << 31;
+const DEFAULT_SOCKET_PATH: &str = "/tmp/security-daemon-socket";
+
+/// Low level client structure to send a `Request` and get a `Response`.
+#[derive(Clone, Debug)]
+pub struct RequestHandler {
+    pub max_body_size: usize,
+    pub timeout: Option<Duration>,
+    pub socket_path: PathBuf,
+}
+
+impl RequestHandler {
+    /// Send a request and get a response.
+    pub fn send_request(&self, request: Request) -> Result<Response> {
+        // Try to connect once, wait for a timeout until trying again.
+        let mut stream =
+            UnixStream::connect(&self.socket_path).expect("Failed to connect to Unix socket");
```

Comment:
I think it would be better to return errors instead of panicking there. `ResponseStatus` implements a [conversion](https://docs.rs/parsec-interface/0.10.0/parsec_interface/requests/enum.ResponseStatus.html#impl-From%3CError%3E) from `std::io::Error` that can be used here I think.

Reply:
See PR description - I won't fix this here, don't intend this to be a fully functional client. We can make these changes in smaller PRs this week

Reply:
I don't totally agree, even if the client is not fully functional in terms of features, the code quality ought to be the same level as in our other repos where we seemed to have banned any kind of panicking.

Reply:
I'd say code correctness and maintaining rigor is important, but at the same level as having review-able PRs. I don't agree that any commit in the tree is required to be fully conformant to the standards that we defined for our releases, simply because it brings no benefit - no one is intended to use the code, so why would it matter if it can panic? I agree with things like variable names, comments, correct formatting since these make development easier, but even business-logic correctness is irrelevant as long as this is understood and tracked. That's why we're ok with shipping Parsec while only supporting a very restricted set of algorithms, for example - a client could rightly claim that it's in the interface, so it should be supported, but it isn't, because we've restricted ourselves to what is feasible and necessary.

Reply:
> having review-able PRs

If replacing the panic would add a lot of complexity to the PR then I agree it would be a problem for the review, it could be noted and done in a later commit. In that case, you would just have to replace the `.expect(...)` with `?`: we would win both conciseness of the PR and future-proof it by preventing the panic.

If you would still prefer to do it later then please file an issue about this topic and I would be fine with it.

Reply:
Yes and no - it would be really easy indeed to change this, maybe even more compact, but given that there's a need to create new structures for error management which would lead to this implicit conversion being changed, using `?` might actually hide discrepancies between how we treat error types. 

Will make this change and add a note in the error issue that needs taking care of! But I stand by my more general point of not enforcing the same level of quality continuously.

Reply:
Done!